### PR TITLE
Feature: Audio Preview

### LINKED ISSUE #364: Feature: Alarm Preview
### Description

When we upload a lot of alarms, at one point, there might be a case where we don't know what a particular tone sounds like. For that reason we will add an alarm preview.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by duckcommit**
### Description
User can now preview the audio before selecting the tone, hence reducing confusion between multiple audios, especially if we add more options from our side.

## Screenshots
https://github.com/CCExtractor/ultimate_alarm_clock/assets/22213675/76752164-39c4-4fe5-a9c6-58bbd896c5ca

## Checklist

- [x] Multiple Tests/Checks
- [x] Documentation has been updated to reflect the changes
- [x] Code follows the established coding style guidelines

### CODE REVIEW

File: lib/app/utils/audio_utils.dart [hunk 1]
```diff
@@ -88,6 +88,65 @@ class AudioUtils {
     }
   }
 
+  static bool isPreviewing = false;
```
Comment (MarkisDev) [inline lib/app/utils/audio_utils.dart:91]: Please move this to the begining of the file, all variables are kept there.

**Review by MarkisDev — CHANGES_REQUESTED** [review]

**Review by duckcommit — COMMENTED** [review]
Yes, I have made that change @MarkisDev

**Comment by MarkisDev** [comment]
@duckcommit  You'll have to resolve the conflicts and recheck the code since it's changed.

**Review by duckcommit — COMMENTED** [review]
The conflicts are now solved. I have tried the code too, it runs fine @MarkisDev .

**Review by MarkisDev — APPROVED** [review]

**Pull request merged**